test/language/expressions/instanceof/primitive-prototype-with-object.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2015 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es6id: 12.9.3
+description: >
+  Throws a TypeError if `prototype` property is not an Object.
+info: >
+  12.9.3 Runtime Semantics: Evaluation
+  RelationalExpression : RelationalExpression instanceof ShiftExpression
+    ...
+    7. Return InstanceofOperator(lval, rval).
+
+    12.9.4 Runtime Semantics: InstanceofOperator(O, C)
+    ...
+    6. Return OrdinaryHasInstance(C, O).
+
+    7.3.19 OrdinaryHasInstance
+    ...
+    3. If Type(O) is not Object, return false.
+    4. Let P be Get(C, "prototype").
+    5. ReturnIfAbrupt(P).
+    6. If Type(P) is not Object, throw a TypeError exception.
+    ...
+---*/
+
+Function.prototype.prototype = "";
+
+assert.throws(TypeError, function() {
+  [] instanceof Function.prototype;
+});
```

Comment:
I think we want to keep the old and the new tests to catch this V8 bug:
```
V8 version 4.9.0 (candidate)
d8> function f(){} f.prototype = 0; [] instanceof f
(d8):1: TypeError: Function has non-object prototype '0' in instanceof check
function f(){} f.prototype = 0; [] instanceof f
                                              ^
TypeError: Function has non-object prototype '0' in instanceof check
    at (d8):1:47

d8> var f = Function.prototype; f.prototype = 0; [] instanceof f
false
```

I'll add these notes to all new tests:
```js
// Check with primitive "prototype" property on non-constructor function.
Function.prototype.prototype = "";

// The "prototype" property for constructor functions is a non-configurable data-property,
// therefore we need to use a non-constructor function to install the getter.
Object.defineProperty(Function.prototype, "prototype", { ...
```